### DISCUSSION TITLE
Folder suddenly not in output anymore, therefore not reachable with process.cwd()

### DISCUSSION THREAD

**Discussion by rubenkaiser** [vercel/community#911] [Help]
In several of our projects we use e-mail templates (with handlebars) to send messages. To enable this we read the .html files containing the handlebars templates using the following code (as Vercel prescribes it):

```
const emailsDir = path.resolve(process.cwd(), 'emails');

const emailFile = readFileSync(path.join(emailsDir, 'template.html'), {
    encoding: 'utf8',
});
```

We've done it this way for quite some time but suddenly one of our projects is failing on Vercel (others are fine). Locally it still works.

When logging process.cwd() we can see that the 'emails' folder is not present at all. So we are quite puzzled why this is suddenly happening. Anyone experiencing similar issues?

Node 14.x
Next 12.3.1

**Comment by amyegan** [comment]
Sorry you're running into this problem @rubenkaiser. Can you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) so I can get a better understanding of what is supposed to be happening and why it is failing?

**Reply by rubenkaiser** [reply]
Actually that is part of the problem. Other projects are identical and still work. In these projects the mentioned 'emails' folder is deployed and accessible. Almost seems like the project is deployed differently by vercel.

I will try again however 👍

**Reply by amyegan** [reply]
Hmm. 🤔 Are there any differences in settings or environment variables that you can identify?

**Reply by rubenkaiser** [reply]
The last change in the code, settings and env vars was 5 months ago. Other projects in the same monorepo work as expected. Tomorrow I will create a separate test, strange as it might sound: hopefully that one will fail as well

**Reply by rubenkaiser** [reply]
After deploying the minimal repoducible example I made the code is working as expected. In this case the 'emails' folder is present in the logs, and accessible for fetching the html files:

https://gitlab.com/rubenkaiser/nextjsfiletest

At this point I'm puzzled to why this is not working in one of our projects on Vercel (we are running several with the same code for emails)

**Reply by amyegan** [reply]
Since we know the email code works in different projects, I think we need to focus on what is different between the broken project and the others that are working as expected. It's possible something is running into an error elsewhere in the app and causing this email problem.

Do you have any logging set up to capture the exact point of failure? If not, I would recommend setting up some [temporary logging](https://vercel.com/docs/concepts/deployments/logs#function-logs) to view the errors in your project dashboard and consider [log drain solutions](https://vercel.com/guides/how-do-i-store-logs-on-vercel#log-drain-solutions) to store the logs.

Are all of your projects using the same tokens for access things like API access? Or is it possible the failing project has and expired token or typo in one of the values?

Are there any warnings coming up during build and deploy? Maybe something that seems unrelated is secretly the source of this problem.

**Reply by rubenkaiser** [reply]
The code does not seem to be the problem. We added logging and could clearly see that in one project the 'emails' folder was present when logging the running directory contents, but in the failing project is wasn't, making it seem like a deployment issue.
